### PR TITLE
Users with Team role can download images

### PR DESCRIPTION
**Pull request by eessex**
- Adds utils `userIsTeam` helper
- Updates all admin checks related to images to determine access by `team` role rather than `admin`
- Adds roles  to `User` typings

Addresses https://artsyproduct.atlassian.net/browse/PLATFORM-2492

### CODE REVIEW

File: src/v2/Utils/user.ts [hunk 1]
```diff
@@ -37,6 +37,17 @@ export function userIsAdmin(user?: User): boolean {
   return isAdmin
 }
 
+export function userIsTeam(user?: User): boolean {
+  const isTeam = Boolean(user && user.type === "Team" ? true : false)
```
Comment (joeyAghion) [inline src/v2/Utils/user.ts:41]: Is it ever the case that `user.type === "Team"`? I think the [`type` property](https://github.com/artsy/gravity/blob/3d89832cca98bd267a06d9caf9f63cc1cffa8c50/app/models/domain/user.rb#L575) is just a legacy thing from when `User`s and `Admin`s were different models. What you might want to check here is `roles`.

**Review by joeyAghion — COMMENTED** [review]
As they're currently designed, `admin`s are purely a subset of `team`, so you might be able to simplify this and only check `isTeam` rather than `isAdminOrTeam` for image-access purposes.

File: src/v2/Utils/user.ts (hunk 1 above)
Comment (eessex) [reply src/v2/Utils/user.ts:41]: Yes this looks correct -- `roles` is not currently available on the user, diging into this further.

Comment (eessex) [reply src/v2/Utils/user.ts:41]: Added this related PR to passport so we can access the `roles` field in Force: https://github.com/artsy/artsy-passport/pull/154

**Comment by eessex** [comment]
Logic has been updated to use `roles` over user type. Depends on https://github.com/artsy/force/pull/6590

File: src/typings/gravity.d.ts [hunk 2]
```diff
@@ -8,3 +9,5 @@ interface User {
   type?: string
   name?: string
 }
+
+type UserRoles = "admin" | "team" | "user" | "genomer"
```
Comment (joeyAghion) [inline src/typings/gravity.d.ts:13]: Is it OK that this isn't the [full list](https://github.com/artsy/gravity/blob/master/config/initializers/_user_roles.rb)?

**Review by joeyAghion — APPROVED** [review]

File: src/typings/gravity.d.ts (hunk 2 above)
Comment (eessex) [reply src/typings/gravity.d.ts:13]: I can go ahead and add these - thanks for this list!

Comment (joeyAghion) [reply src/typings/gravity.d.ts]: But does it _have_ to match, or does this just need to include the subset that Force's code references? I wouldn't want the addition of a new role to become runtime errors.

Comment (eessex) [reply src/typings/gravity.d.ts]: I'm pretty sure we are safe on this but will monitor.

**Pull request merged**